src/component/main/LongRunningTasksStatus.tsx

```diff
@@ -0,0 +1,155 @@
+  const runningTasks = useSelector(
+    (state: TermItState) => state.runningTasks,
+    isEqual
+  );
+  const { i18n } = useI18n();
+
+  const tasksToUpdate = useMemo<Set<() => void>>(() => new Set(), []);
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      tasksToUpdate.forEach((task) => task());
+    }, 250);
+    return () => clearInterval(timer);
+  }, [tasksToUpdate]);
+
+  if (Object.keys(runningTasks).length === 0) {
+    return null;
+  }
+
+  return (
+    <UncontrolledDropdown nav={true} className="long-running-tasks-dropdown">
+      <DropdownToggle
+        nav={true}
+        caret={false}
+        className={classNames(
+          "text-dark text-dropdown",
+          "d-flex",
+          "align-items-center",
+          "long-running-tasks-toggle"
+        )}
+      >
+        <Mask classes="align-middle" withoutText={true} />
+      </DropdownToggle>
+      <DropdownMenu
+        className="dropdown-menu-arrow long-running-tasks-menu"
+        right={true}
+      >
+        <DropdownItem
+          text={true}
+          className="dropdown-item-text long-running-tasks-label"
+        >
+          <p>{i18n("longrunningtasks.description")}</p>
+        </DropdownItem>
+
+        {Object.keys(runningTasks).map((uuid, i) => {
+          const task = runningTasks[uuid];
+          return (
+            <DropdownItem
+              key={"long-running-task-" + uuid + i}
+              text={true}
+              className="long-running-tasks-task"
+            >
+              {icon(task, i18n)}
+              <span>{i18n(task.name)}</span>
+              <If
+                expression={
+                  !!task.startedAt &&
+                  task.state === LongRunningTaskState.RUNNING
+                }
+              >
+                <span className="text-muted">
+                  <DurationDisplay
+                    since={task.startedAt!}
+                    tasksToUpdate={tasksToUpdate}
+                  />
+                </span>
+              </If>
+            </DropdownItem>
+          );
+        })}
+      </DropdownMenu>
+    </UncontrolledDropdown>
+  );
+};
+
+const DurationDisplay: React.FC<{
+  since: Date;
+  tasksToUpdate: Set<() => void>;
+}> = ({ since, tasksToUpdate }) => {
+  const tZero = useMemo(() => {
+    if (since.getTime() > Date.now()) {
+      return Date.now();
+    }
+    return since.getTime();
+  }, [since]);
+
+  const [time, setTime] = useState(0);
+  useEffect(() => {
+    const task = () => {
+      setTime(Date.now() - tZero);
+    };
+    tasksToUpdate.add(task);
+
+    return () => {
+      tasksToUpdate.delete(task);
+    };
+  }, [tZero, tasksToUpdate]);
+
+  const date = new Date(time);
+  const hours = date.getUTCHours() > 0 ? date.getUTCHours() + ":" : "";
+  const minutes = padStart(date.getUTCMinutes().toString(), 2, "0");
+  const seconds = padStart(date.getUTCSeconds().toString(), 2, "0");
+
+  return <>{`${hours}${minutes}:${seconds}`}</>;
```

Comment:
What about using TimeAgo to display the duration of the task? We already use for displaying time since last edit/comment.

Reply:
I wanted to show how long the task has been running.